openmc/tallies.py

```diff
@@ -21,9 +22,58 @@ def reset_auto_tally_id():
 
 
 class Tally(object):
+    """A tally defined by a set of scores that are accumulated for a list of
+    nuclides given a set of filters.
+
+    Parameters
+    ----------
+    tally_id : int
+        Unique identifier for the tally
+    name : str
+        Name of the taly
+
+    Attributes
+    ----------
+    id : int
+        Unique identifier for the tally
+    name : str
+        Name of the tally
+    filters : list of openmc.filter.Filter
+        List of specified filters for the tally
+    nuclides : list of openmc.nuclide.Nuclide
+        List of nuclides to score results for
+    scores : list of str
+        List of defined scores, e.g. 'flux', 'fission', etc.
+    estimator : {'analog', 'tracklength'}
+        Type of estimator for the tally
+    triggers : list of openmc.trigger.Trigger
+        List of tally triggers
+    num_score_bins : int
+        Total number of scores, accounting for the fact that a single
+        user-specified score might have multiple bins
```

Comment:
I might just add something like "(e.g., 'scatter-P3', 'flux-Y2,2')" in case some users find this concept confusing (I did when I began using the code)